scripts/check_types.py

```diff
@@ -0,0 +1,104 @@
+"""
+Type hints checker using mypy.
+"""
+
+import json
+import logging
+import subprocess
+import sys
+import tempfile
+from collections import defaultdict
+from pathlib import Path
+
+# Set up logging
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+logger = logging.getLogger(__name__)
+
+
+def main():
+    # Setup
+    script_dir = Path(__file__).resolve().parent
+    dirs = sys.argv[1:] or [
+        script_dir.parent / "prolif",
+        script_dir.parent / "tests",
+    ]
+    dirs = [d for d in dirs if Path(d).exists()]
```

Comment:
can you cast each item to be a `Path` object directly? You can use the walrus operator (`:=`) to avoid casting to a Path twice